src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/ApplicationsController.kt

```diff
@@ -117,14 +115,18 @@ class ApplicationsController(
 
     val applicationResult = when (xServiceName ?: ServiceName.approvedPremises) {
       ServiceName.approvedPremises -> applicationService.createApprovedPremisesApplication(personInfo.offenderDetailSummary, user, deliusPrincipal.token.tokenValue, body.convictionId, body.deliusEventNumber, body.offenceId, createWithRisks)
-      ServiceName.cas2 -> applicationService.createCas2Application(body.crn, user, deliusPrincipal.token.tokenValue)
       ServiceName.temporaryAccommodation -> {
         when (val actionResult = applicationService.createTemporaryAccommodationApplication(body.crn, user, deliusPrincipal.token.tokenValue, body.convictionId, body.deliusEventNumber, body.offenceId, createWithRisks)) {
           is AuthorisableActionResult.NotFound -> throw NotFoundProblem(actionResult.id!!, actionResult.entityType!!)
           is AuthorisableActionResult.Unauthorised -> throw ForbiddenProblem()
           is AuthorisableActionResult.Success -> actionResult.entity
         }
       }
+
+      ServiceName.cas2 -> throw RuntimeException(
+        "CAS2 now has its own " +
+          "Cas2ApplicationsController",
+      )
```

Comment:
Probably overkill, but could we redirect here?

Reply:
Hmm, possibly?  

At present I'm thinking that this is very much an interim pathway which exists only during the refactoring.  It's intended to handle the scenario where the UI is POSTing to `/applications` rather than to the new `cas2/applications` endpoint and give a helpful error message to developers.